### DISCUSSION TITLE
Running a simulation intermittently

### DISCUSSION THREAD

**Discussion by oferst13** [pyswmm/pyswmm#383] [Q&A] [answered]
Hello!
We need to run simulations intermittently in the following way:
1. Run a sim
2. Do some optimizations outside of the pyswmm simulation
3. Run a sim for a shorter period than stage 1, with different rainfall but with the same init conditions
4. Back to stage 1 with new init conditions set by stage 3 (There is some temporal overlap between each sequence)

Is there a way to use sim.initial_conditions for the entire model with the values from an output file without coding them explicitly?

**Comment by bemcdonnell** [comment, marked as answer]
@oferst13, it really sounds like you could try to use a hotstart file for that. Are you familiar with that?  That would take care of all of the initial conditions for you and allow you to use a different rainfall input.

**Reply by oferst13** [reply]
Very little documentation on the topic but I think I understand how it works in SWMM, but is there a way to save and use .hsf files dynamically using pyswmm so I won't have to create many .inp files each with its own .hsf file?
I use raingauge.total_precip = ... for each timestep to set the rainfall from another source, so the rainfall is less of an issue.
And thank you very much!

**Reply by bemcdonnell** [reply]
@oferst13 we started a new feature years ago related to this:

https://github.com/OpenWaterAnalytics/Stormwater-Management-Model/pull/252

It was called “breadcrumbs” (named after Little Red Riding Hood who left breadcrumbs in the forest so she would know where to return back to.)  We never finished it. If you want to take it across the finish line and then extend the features into pyswmm, you’d be a hero!

**Reply by oferst13** [reply]
Well, I barely succeed in getting my python code to work.. I guess it's hard finding contributors to a opensource water engineering project.
Anyway, I found out that if the model is run with pyswmm it can read and write .hsf files if they are defined in SWMM, so maybe I'll have two .inp files - one for stage 1 which reads .hsf and one for stage 3 which writes them. Thanks again for a great project!

P.S It was Hansel and Gretel who used breadcrumbs, and it didn't go well for them..

**Reply by bemcdonnell** [reply]
@oferst13, oh that gave me a good laugh today 😂🤣😂!  And now I remember that the little red riding hood made a different set of mistakes! 

As for pyswmm, as you identified, you can pre-process the INP file to Save and Use the hot start.  We are introducing a whole set of tutorial videos on our website (about 15% complete). Check out https://www.pyswmm.org

**Reply by dickinsonre** [reply]
**I signed up for the newsletter for https://www.pyswmm.org/ - well done website**

Hot start files in any software are saved at the end of a run and used at the beginning of the run.   In SWMM generally, as long as you have the same number of nodes, links etc then any created hsf file can be used by any other run

**Reply by oferst13** [reply]
Thanks @dickinsonre @bemcdonnell, looks like a good solution for my program. I'll spread the word about pyswmm.org.
Ofer
